src/transports/httpclient.c

```diff
@@ -734,7 +734,7 @@ static int generate_request(
 		git_buf_printf(buf, "Expect: 100-continue\r\n");
 
 	if ((error = apply_server_credentials(buf, client, request)) < 0 ||
-	    (error = apply_proxy_credentials(buf, client, request)) < 0)
+	    (!client->proxy_connected && (error = apply_proxy_credentials(buf, client, request)) < 0))
```

Comment:
Well I'll be damned. Yea, that totally works in testing. In fact, it works better than what I had originally wrote in that the proxy authentication is _never_ sent to the git provider (where as in both the _old code and current flow in this PR_ send the proxy-auth to the git server on first `GET`).

Reply:
I see. Thanks for explaining that. I understand and agree with your take on this. We leverage our proxy credentials in a connect proxy in generate_connect_request, and thus don't need to ever add them in this method when we are a connect proxy. This makes sense.

Reply:
Apologies for the ping, I edited that last one and want to make sure that you get the notification @ethomson.

Reply:
> Are you certain that the approach I took will have a regression for non-CONNECT proxies?

🤔  No, I'm not certain.  I'm armchair reviewing from my phone.  🙃

I appreciate you explaining this, I think that you're right and that your approach should work.  I had trouble reasoning about `proxy_connected`.

I'd like to spin up some tests to play with this more but I wouldn't have too many hesitations about shipping hours as if if I were you.  Worst case scenario: http (not https) auth breaks to git servers through a proxy.  I'm actually not even sure that we support auth over http at the layer above this.

Reply:
>  I had trouble reasoning about `proxy_connected`.

Mutable state was easy to reason about? :joy:

I am going to take your solution, because it ultimately relies not on whether a connection was ever established, but on the fact that if one was ever established, that check must also hold. I prefer the non mutable approach because it's much easier to reason about.

Thanks for the prompt reply, that helped us make a decision.